subprojects/model-core/src/integTest/groovy/org/gradle/api/internal/model/CustomPluginObjectFactoryIntegrationTest.groovy

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.internal.model
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class CustomPluginObjectFactoryIntegrationTest extends AbstractIntegrationSpec {
+    def "plugin can create unnamed instances of class using injected factory"() {
```

Comment:
The code might be easier to manage and parse if we'd move the classes into `buildSrc`.